[RNMobile] Upgrade to RN 0.64

dratwas opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

## Description
Upgrade React Native to version 0.64.0 (See [RN Upgrade Helper](https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.64.0))

Related PRs:

- Gutenberg-Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3430
- WPiOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/16562
- WPAndroid: https://github.com/wordpress-mobile/WordPress-Android/pull/14702
- react-native-svg PR: https://github.com/wordpress-mobile/react-native-svg/pull/12
- react-native-keyboard-aware-scroll-view PR: https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/12
- react-native-gesture-handler PR: https://github.com/wordpress-mobile/react-native-gesture-handler/pull/1
- react-native-video: https://github.com/wordpress-mobile/react-native-video/pull/10
- react-native-linear-gradient: https://github.com/wordpress-mobile/react-native-linear-gradient/pull/1
- react-native-hsv-color-picker: https://github.com/wordpress-mobile/react-native-hsv-color-picker/pull/1
- react-native-safe-area-context upgrade PR: https://github.com/wordpress-mobile/react-native-safe-area-context/pull/2
- react-native-slider: https://github.com/wordpress-mobile/react-native-slider/pull/6
- react-native-get-random-values: https://github.com/wordpress-mobile/react-native-get-random-values/pull/1
- react-native-masked-view: https://github.com/wordpress-mobile/react-native-masked-view/pull/2
- react-native-screens: https://github.com/wordpress-mobile/react-native-screens/pull/2
- react-native-reanimated: https://github.com/wordpress-mobile/react-native-reanimated/pull/2
- react-native-prompt-android: https://github.com/wordpress-mobile/react-native-prompt-android/pull/3

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
To test: Gutenberg Editor should work as expected.
Try to run all the manual tests.

Remaining Tasks:
- [x] `rnmobile/try/upgrade-0-64` branch is updated to 0.64.0-rc3 and needs to be updated to 0.64.0 stable. 
- [x] `rnmobile/try/upgrade-0-64` needs to be updated with latest gutenberg trunk which contains some larger updates for handling WPAndroid composite builds
- [x] iOS Demo App needs to be fixed on `rnmobile/try/upgrade-0-64`
- [x] Need to create relevant PRs in other Repos following model of https://github.com/wordpress-mobile/gutenberg-mobile/pull/2580
- [x] Re-target to [Gutenberg Web React 17 update branch](https://href.li/?https://github.com/WordPress/gutenberg/pull/26847), or bring those changes in here
- [x] Check CI across all repos and address any issues

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/native-mobile.md -->

SergioEstevao committed:
Make sure that _input exists.

SergioEstevao committed:
Fix HtmlTextInput tests.

Use accessibility label to find TextInput component instead of the class
name. The class name fails because the way TextInput is implemented in
0.62 where a ForwardRef is used.

SergioEstevao committed:
Fix TextInput mock.

Add isFocused method to mock of TextInput.

marecar3 committed:
fix lint errors

marecar3 committed:
Bump rn to 0.62.2

marecar3 committed:
Remove hack as PR fix: https://github.com/facebook/react-native/pull/28332 is release with 0.62.2

marecar3 committed:
[iOS] Use react native 0.62 hook for dark mode (#21455)

marecar3 committed:
Bump react to 16.11.0

marecar3 committed:
Use correct version of react and react-dom

aduth committed:
Framework: Update package-lock.json

Changes after install on clean clone

change

marecar3 committed:
Update package-lock.json

marecar3 committed:
fix package.json

gziolo committed:
Apply local changes to package-lock.json

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

# Conflicts:
#	package-lock.json

marecar3 committed:
Update license checker

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

# Conflicts:
#	package-lock.json

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

marecar3 committed:
Merge

marecar3 committed:
Merge branch 'master' into rnmobile/update-rn-0-62

SergioEstevao committed:
Merge branch 'master' into rnmobile/update-rn-0-62

# Conflicts:
#	package-lock.json

SergioEstevao committed:
Update packages to use RN version 0.62.2

SergioEstevao committed:
Merge branch 'master' into rnmobile/update-rn-0-62

# Conflicts:
#	package-lock.json
#	packages/react-native-editor/ios/Podfile.lock
#	packages/react-native-editor/package.json
#	packages/scripts/scripts/check-licenses.js

SergioEstevao committed:
Update project locks.

SergioEstevao committed:
Merge branch 'master' into rnmobile/update-rn-0-62

# Conflicts:
#	packages/react-native-editor/ios/Podfile.lock

SergioEstevao committed:
Match babel-preset between react-native-editor and gutenberg packages.

SergioEstevao committed:
Update lock file.

SergioEstevao committed:
Merge branch 'master' into rnmobile/update-rn-0-62

# Conflicts:
#	packages/react-native-editor/ios/Podfile.lock
#	packages/react-native-editor/package.json

SergioEstevao committed:
Fix package number.

SergioEstevao committed:
Update podfile.lock

ceyhun committed:
Merge branch 'master' into rnmobile/update-rn-0-63

ceyhun committed:
Update react-native-svg and react-native-video refs

ceyhun committed:
Update react-native patch to 0.62.2

ceyhun committed:
Upgrade react-native and related packages

ceyhun committed:
Update pods

ceyhun committed:
Add back metro-react-native-babel-transformer

ceyhun committed:
Update react-native-keyboard-aware-scroll-view ref

ceyhun committed:
Update RN version in build.gradle

ceyhun committed:
Hoist `jsc-android` to top-level node_modules folder

Also added `--no-jetifier` flag to "react-native run-android" as it invokes
jetifier which searches for a node_modules folder under packages/react-native-editor
and throws an error when it couldn't find it.

ceyhun committed:
Remove metro patch that's no longer needed

ceyhun committed:
Remove flex-wrap from contentContainerStyle of KeyboardAwareFlatList

This fixes the console error:
`flexWrap: `wrap`` is not supported with the `VirtualizedList` components.
Consider using `numColumns` with `FlatList` instead.

ceyhun committed:
Use correct key prop in picker render method

This fixes the console error:
Each child in a list should have a unique "key" prop.

ceyhun committed:
Use native ref instead of tag to focus and blur

This fixes the console error:
focusTextInput must be called with a host component. Passing a react tag is deprecated.

ceyhun committed:
Replace deprecated currentlyFocusedField with currentlyFocusedInput

This fixes the console error:
currentlyFocusedField is deprecated and will be removed in a future release. Use currentlyFocusedInput

ceyhun committed:
Update react-native-keyboard-aware-scroll-view ref

ceyhun committed:
Revert "Replace deprecated currentlyFocusedField with currentlyFocusedInput"

This reverts commit ace62184be1bcd56c5785bf3ca16be0b9317e809.

ceyhun committed:
Revert "Use native ref instead of tag to focus and blur"

This reverts commit 47bd77120e6160e2c4aae3a69228897f4581b3e9.

ceyhun committed:
Use old implementation of TextInputState in AztecView

The implementation is from here:
https://github.com/facebook/react-native/blob/e1cfeaddd47149625cc8c76c9b50c09b4c19e296/Libraries/Components/TextInput/TextInputState.js

ceyhun committed:
Remove RN 0.62 patch

ceyhun committed:
Add RN 0.63 patch

ceyhun committed:
Revert "Use old implementation of TextInputState in AztecView"

This reverts commit f0ea6848409f634516d051d495a7ad3a4a63432a.

ceyhun committed:
Merge branch 'master' into rnmobile/update-rn-0-63

ceyhun committed:
Merge branch 'master' into rnmobile/update-rn-0-63

ceyhun committed:
Hoist jsc-android to top-level node_modules folder again after merging master

ceyhun committed:
Update cli-platform-ios path

ceyhun committed:
Fix iOS focus/blur crashes via implementing them in native and using ref instead of findNodeHandle

ceyhun committed:
Use ref instead of innerRef for react-native-keyboard-aware-scroll-view

ceyhun committed:
Merge branch 'master' into rnmobile/update-rn-0-63

ceyhun committed:
Update package-lock.json

ceyhun committed:
Fix bundle commands

ceyhun committed:
Update Podfile.lock

ceyhun committed:
Update unit test snapshot

ceyhun committed:
Fix iOS build cache hit test

ceyhun committed:
Bust iOS build cache

ceyhun committed:
Revert "Bust iOS build cache"

This reverts commit 96c11e93c53606e142db4a1869594483c6009d92.

youknowriad committed:
Update to React 17.0.1

youknowriad committed:
Update to React 17.0.1

ceyhun committed:
Merge branch 'master' into rnmobile/update-rn-0-63

ceyhun committed:
Bust iOS build cache

ceyhun committed:
Fix hermes CLI name

ceyhun committed:
Remove iOS build cache

ceyhun committed:
Update package-lock.json

ceyhun committed:
Update remaining android files from 0.63 diff

ceyhun committed:
Fix android focus/blur issues

youknowriad committed:
Update test utilities

youknowriad committed:
Update test utilities

cameronvoell committed:
Replace deprecated currentlyFocusedField

cameronvoell committed:
Prevents immediate crash on physical android devices

youknowriad committed:
Suppress console warnings when unmounting refs

youknowriad committed:
Suppress console warnings when unmounting refs

dratwas committed:
update deps

dratwas committed:
merge with master

dratwas committed:
update dependencies

dratwas committed:
update dependencies in react-native-editor

dratwas committed:
merge with master - resolve conflicts

dratwas committed:
remove patch for rn 0.63.2

dratwas committed:
update package-lock.json to resolve errors about 2 versions of react installed

WunderBart committed:
Update enzyme adapter import

WunderBart committed:
Update enzyme adapter import

WunderBart committed:
Fix failing Sandbox spec

WunderBart committed:
Fix failing Sandbox spec

diegohaz committed:
Upgrade Reakit version

diegohaz committed:
Upgrade Reakit version

cameronvoell committed:
Update react-native to 0.64.0 stable

cameronvoell committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

vdwijngaert committed:
Add workaround for enzyme adapter unmount issue

vdwijngaert committed:
Add workaround for enzyme adapter unmount issue

vdwijngaert committed:
Updates snapshots to reflect changes to reakit handling popovers

vdwijngaert committed:
Updates snapshots to reflect changes to reakit handling popovers

ceyhun committed:
Upgrade cocoapods gem to 1.10.1

ceyhun commented on package.json:
```diff
@@ -80,7 +80,9 @@
-		"@wordpress/wordcount": "file:packages/wordcount"
+		"@wordpress/wordcount": "file:packages/wordcount",
+		"fbjs": "3.0.0",
+		"react-native-video": "git+https://github.com/wordpress-mobile/react-native-video.git#97792c7737b234fe48840d443a9b3135711e5df7"
```
I think these should be in `devDependencies` and could be even better to have them in `react-native-editor/package.json` if possible.

ceyhun committed:
Add workaround for https://github.com/facebook/react-native/issues/31034

ceyhun commented:
@cameronvoell Took a quick look and attempted `npm run native ios` to begin. I'm using Xcode 12.4. 

The first error was during `bundle exec pod install` phase where a greater version of CocoaPods was needed. I upgraded it to 1.10 and pushed the changes.

Then I got another error during build phase in Xcode:

<img width="600" src="https://user-images.githubusercontent.com/1845482/116595256-6e098800-a923-11eb-92c5-a81bee5aed48.png">

Adding `#import <React/RCTEventDispatcher.h>` in this file resolved that issue. 
Warning ⚠️ We still need to commit this change to our `react-native-gesture-handler` fork and update its ref in this PR  

Then I got a `Cycle inside FBReactNativeSpec` error. It seems to be this issue: https://github.com/facebook/react-native/issues/31034. Using [this](https://github.com/facebook/react-native/issues/31034#issuecomment-812564390) proposed solution, I was able to resolve it. I also pushed this change.

After that the iOS demo app built successfully for me 🎉

cameronvoell commented:
> After that the iOS demo app built successfully for me 🎉

@ceyhun Nice! That's great, I only got to test this for a short bit end of day, and was seeing an error, so guessing it's something with my local env since it worked for you. Let me know if you have any ideas, otherwise I spend some more time looking.

Xcode version is 12.5. My error looked like this:
<details>

`npm run native ios` output:
```
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/cameronvoell/Library/Developer/Xcode/DerivedData/GutenbergDemo-eqaoziwexisoucddwaicvmipdjoc/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNGestureHandler.build/Objects-normal/x86_64/RNGestureHandlerManager.o /Users/cameronvoell/A8C/DEVELOP/gutenberg/node_modules/react-native-gesture-handler/ios/RNGestureHandlerManager.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(6 failures)
```

In Xcode:
![image](https://user-images.githubusercontent.com/1103838/116657583-85586d80-a943-11eb-84ce-54d926b6f93f.png)


</details>

ceyhun commented:
@cameronvoell That seems to be the same error I shared above (only the screenshot is in light mode) 😄 Please check my comment above to resolve it.

hypest commented:
Headsup that [React Native v0.64.1 was released yesterday that includes a fix for Xcode 12.5](https://github.com/facebook/react-native/releases/tag/v0.64.1).

ceyhun commented:
Another heads up :) Xcode 12.5 requires macOS 11, which is in [private preview](https://github.com/actions/virtual-environments/issues/2486) on GitHub Actions and not available on CircleCI.

cameronvoell committed:
Fix for react-native-gesture-handler, missing import

cameronvoell commented:
Hi @ceyhun , I got the ios demo app running from Xcode (still seeing errors when runnning `npm run native ios` command) but the editor is displaying only the inserter and menu options with the Block-list missing. I'm on Xcode 12.4 and testing on a iPhone 11 simulator. Can you confirm if you are seeing the same? 

| visual editor | html editor |
| -------- | -------- |
| <img src="https://user-images.githubusercontent.com/1103838/117479157-d56f9b00-af14-11eb-9cf6-6132a6f43b43.png"  width="400"> |<img src="https://user-images.githubusercontent.com/1103838/117479979-d523cf80-af15-11eb-978a-6ddd69cb02c8.png"  width="400"> |

ceyhun committed:
Remove top-level dependencies

ceyhun committed:
Downgrade react types

React 17 types are conflicting with @storybook react types which are
using version 16. Will keep 16 until we @storybook types as well.

ceyhun commented:
`npm run native ios` seems to throw an error for me as well. But it runs `react-native run-ios` in the end and 
the underlying command `xcodebuild -workspace GutenbergDemo.xcworkspace -configuration Debug -scheme GutenbergDemo -destination id=...` 
works fine for me. Same for building it directly from Xcode. But then the white screen you shared is there for me too.

I tried to move `dependencies` in `package.json` to `packages/react-native-editor/package.json` first.
The error I'm currently getting is in the script phase `[CP-User] Generate Specs` of `FBReactNativeSpec` dependency:

```
nvm is not compatible with the "npm_config_prefix" environment variable: currently set to "....gutenberg/packages/react-native-editor"
Run `unset npm_config_prefix` to unset it.
```

I tried adding `. "$HOME/.nvm/nvm.sh"` to the top of the `Bundle React Native code and images` script phase of GutenbergDemo target, but that didn't fix the issue.

I pushed the `package.json` changes to this branch. I'll keep looking into this.

ceyhun committed:
Fix Xcode shell script `Bundle React Native code and images`

ceyhun committed:
Add workaround for another iOS build error

The error is during `npm run native ios`:
nvm is not compatible with the "npm_config_prefix" environment variable

ceyhun committed:
Changes after running pod install

ceyhun committed:
Update react-native-svg ref

ceyhun committed:
Re-install @wordpress/react-native-editor

ceyhun committed:
Changes after running pod install

ceyhun committed:
Update react-native-keyboard-aware-scroll-view ref

ceyhun commented:
I fixed the `unset npm_config_prefix` error with another workaround in the Podfile. After that `npm run native ios` started working again :tada:

But in the simulator I started seeing errors saying `css-to-react-native-transform` is missing. Which is a dependency of `react-native-sass-transformer`, which was missing. I tried to re-install `@wordpress/react-native-editor` to bring back the missing dependencies using these commands:
```
npm uninstall @wordpress/react-native-editor --save && npm install ./packages/react-native-editor --save
``` 

But during the `npm install`, I got multiple TypeScript errors related to the build of `react-native-svg`. I realised that because of those errors, `@wordpress/react-native-editor` wasn't properly installed before as well. 

So I went back to our fork of `react-native-svg` to resolve this issue. Reverting the `@types/react-native` back to what it was before, fixed it. Created [a PR](https://github.com/wordpress-mobile/react-native-svg/pull/12) for that change and linked it in the description here.


After these changes, the white screen issue was still there. I saw in metro logs a warning mentioning `KeyboardAwareFlatList`:
```
Warning: Functions are not valid as a React child. This may happen if you return a Component instead of <Component /> from render.
``` 


After checking `react-native-keyboard-aware-scroll-view`, I came across [this fix](https://github.com/APSL/react-native-keyboard-aware-scroll-view/pull/442). I applied the same change in our fork and created [a PR](https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/12) and linked in the description again. Now the editor is back again 🎉🎊🥳🎆    

I didn't have a chance to properly test if everything is working yet, but I noticed there are some dependencies in `/packages/react-native-editor/node_modules` folder that are not hoisted (`metro-react-native-babel-preset` for example), because they are defined in more than one `package.json` with different versions. It would be a good idea to try to unify them under the top-level `node_modules` folder by having them only in a single `package.json`.

cameronvoell committed:
Update react-native-gesture-handler to fix android build issue

cameronvoell commented:
I noticed today that the Android demo build was broken after we updated the `react-native-gesture-handler`, but I found a fix for that on another [branch](https://github.com/wordpress-mobile/react-native-gesture-handler/tree/upgrade-to-64) from @dratwas (thanks!). On that branch `react-native-gesture-handler` was caught up with upstream fork, and I noticed it built the ios demo without requiring the `#import <React/RCTEventDispatcher.h>` update. 

~~The only issue I'm seeing now is a an issue with focusing on RichText in Android (I know we've had to fix similar bugs in the past, but couldnt find a quick easy fix yet). The error that shows up when clicking on text on Android and iOS is :
I think the only place `focusTextInput` is called is [here](https://github.com/WordPress/gutenberg/blob/fbf0eb266d5a55b420cf7be6e4cc449a64ff8307/packages/react-native-aztec/src/AztecView.js#L187). Maybe you have some ideas on how to fix the focus in text issue on Android?~~ I found where I [accidentally overrode](https://github.com/WordPress/gutenberg/pull/29118/commits/da3608194d074bf918cb11f5cd236c5b5a28139b#diff-166bf22d90bde8c6ae37050a20ead06a22606d7f81ad6e48963269e0736e3f83) a previous fix while resolving conflicts during catching up with trunk. I [fixed that](https://github.com/WordPress/gutenberg/pull/29118/commits/41968e2bacdf7971606dd09d3abfb148d1c15181) and now Android demo app is looking good!

cameronvoell committed:
Updating gesture-handler again after updating to 0.64.0 stable

cameronvoell committed:
Re-added focus/blur fix that was overridden during last update with trunk

cameronvoell committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

cameronvoell committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Changes after running npm install

ceyhun committed:
Fix hermes cli path

ceyhun committed:
Remove iOS npm cache

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Delete all other workflows

ceyhun committed:
Debug CI

ceyhun committed:
Move packages under packages/react-native-editor/node_modules to top-level

ceyhun committed:
Add `--no-jetifier` and `--no-packager` params to "react-native run-android"

Otherwise it invokes jetifier which searches for a node_modules folder under
packages/react-native-editor and throws an error when it couldn't find it.

ceyhun committed:
Revert "Debug CI"

This reverts commit 141fcff5867d5e6f832a6707cb6b484504b3e664.

ceyhun committed:
Revert "Delete all other workflows"

This reverts commit 96b2bc388c1fc69835003b0d93d4f8c017c1adcc.

ceyhun committed:
Upload npm logs as artifacts

ceyhun committed:
Add @react-native to transformIgnorePatterns in jest config

Expecting this to fix the following error:
```
/home/runner/work/gutenberg/gutenberg/node_modules/@react-native/polyfills/error-guard.js:14
type ErrorHandler = (error: mixed, isFatal: boolean) => void;
        ^^^^^^^^^^^^

SyntaxError: Unexpected identifier

    at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1350:14)
    at Object.<anonymous> (node_modules/react-native/jest/setup.js:443:6
```

ceyhun committed:
Add patch to disable metro watch

cameronvoell committed:
fix location for NativeAnimatedHelper

ceyhun committed:
Revert "Add patch to disable metro watch"

This reverts commit d00142381e820335ccdeaad3800b918e11649f7f.

ceyhun committed:
Try force cleaning npm cache

ceyhun committed:
Mock RNGestureHandlerModule manually

ceyhun committed:
Upgrade react-test-renderer to 17.0.1

ceyhun committed:
Install and use @wojtekmaj/enzyme-adapter-react-17

ceyhun committed:
Re-add core-js using the same version in @wordpress/babel-preset-default

ceyhun committed:
Add workaround for https://github.com/facebook/react-native/issues/31034

cameronvoell committed:
Remove temp directory from s3 for hermes / RN binaries

cameronvoell committed:
Updated s3 react-native-mirror location

cameronvoell committed:
Updated android binary locations

cameronvoell committed:
Update sleep time after aztec publish from 30 sec to 2 min

cameronvoell committed:
Adding s3 maven directory

cameronvoell committed:
Remove react-native-mirror frmo gradle file

ceyhun committed:
Fix lint error

cameronvoell committed:
Updated react native libs refs

cameronvoell committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

cameronvoell committed:
Try npm cache verify command to fix ios CI

cameronvoell committed:
Set sleep time after rn-aztec publish back to 30 sec

cameronvoell committed:
Added npm logs as artifacts for android ci job

ceyhun committed:
Debug CI

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Remove Xcode 12.5 RN patch

ceyhun committed:
Use same core-js version with @wordpress/babel-preset-default

ceyhun committed:
Try npm install

ceyhun committed:
Try npm install

ceyhun committed:
Revert "Debug CI"

This reverts commit b4b933305f1bb4ecf49e55b1a7ae8d6668933ed0.

ceyhun committed:
Revert "Try npm cache verify command to fix ios CI"

This reverts commit c6c9344919ba71a04979e50b21bee07df4000048.

ceyhun committed:
Revert "Try force cleaning npm cache"

This reverts commit 91d23825eb6a6ade8270daec312687d55dd37379.

ceyhun committed:
Revert "Remove iOS npm cache"

This reverts commit ac3fe9a47f9287b80b21ecd6196ceac31075cfa1.

ceyhun committed:
Revert "Upload npm logs as artifacts"

This reverts commit b829ceebe8f1ca417f457d30c6854feae008049d.

ceyhun committed:
Trigger CI

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Try removing react-native-prompt-android package

ceyhun committed:
Remove other workflows

ceyhun committed:
Debug CI

ceyhun committed:
Debug CI

ceyhun committed:
Revert "Remove other workflows"

This reverts commit a6d201d5e342b43742ae89375be21c06b45c7658.

ceyhun committed:
Revert "Debug CI"

This reverts commit 61c7e9c5c8dbb981a40bcf3b811fc9b95607d58f.

ceyhun committed:
Revert "Debug CI"

This reverts commit 1006714fa492299bb6e027a97d77bccd7cc745f0.

ceyhun committed:
Try removing react-native-safe-area-context package

ceyhun committed:
Try deleting ~/.npm folder

ceyhun committed:
Revert "Try deleting ~/.npm folder"

This reverts commit 2efc1c87791ba12024d42d44a2d56a3866538305.

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Try node 16

ceyhun committed:
Revert "Try removing react-native-safe-area-context package"

This reverts commit cc47236b78998a29768492eeda675b21b1b90b8c.

ceyhun committed:
Revert "Try removing react-native-prompt-android package"

This reverts commit 9c21f8a534e4601e0e68a2d7ad63823c543a04ff.

ceyhun committed:
Revert "Try node 16"

This reverts commit b4d8a5c8866d33e598b7a1bd655c9b0721931ab4.

ceyhun committed:
Try node 16 in every workflow

ceyhun committed:
Clear npm caches in all workflows

ceyhun committed:
Try without rm -rf ~/.npm

ceyhun committed:
Disable caching ~/.npm in all workflows

ceyhun committed:
Try node 14 and only 'rm -rf ~/.npm'

ceyhun committed:
Try node 14 && 'rm -rf ~/.npm' && 'npm cache clean -force'

ceyhun committed:
Try node 16 && 'npm install -g npm@6' && 'rm -rf ~/.npm'

ceyhun committed:
Try node 16 && 'npm install -g npm@6' && 'npm cache clean -force' && 'rm -rf ~/.npm'

ceyhun committed:
Try node 16 && 'npm cache clean -force' && 'rm -rf ~/.npm' && 'npm install -g npm@6'

ceyhun committed:
Try node 16 && 'npm cache clean -force' && 'npm install -g npm@6' && 'npm cache clean -force' && 'rm -rf ~/.npm'

ceyhun committed:
Try 'nvm install' && 'npm cache clean -force' && 'rm -rf ~/.npm'

ceyhun committed:
Update nvm install command

ceyhun committed:
Print node, npm version

ceyhun committed:
nvm install 16

ceyhun committed:
Add --loglevel verbose to npm ci and install

ceyhun committed:
Try default shell

ceyhun committed:
Remove default shell

ceyhun committed:
Try nvm with node 14

ceyhun committed:
Try adding back cache clean

ceyhun committed:
Try nvm, node 16, npm 6

ceyhun committed:
Revert github workflow changes

ceyhun committed:
Disable node_modules caches

ceyhun committed:
Re-create package-lock.json

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Revert package-lock.json to trunk

ceyhun committed:
Try removing packages with package.json files that have changes

ceyhun committed:
Re-add packages with package.json files that have changes

ceyhun committed:
Try npm cache verify and --loglevel verbose

ceyhun committed:
Revert "Try npm cache verify and --loglevel verbose"

This reverts commit 781e403569212ccaf2da2aa47a01e6cfb2787a54.

ceyhun committed:
Use node 12 instead of 14

ceyhun committed:
Revert "Use node 12 instead of 14"

This reverts commit 6f72d0b57e1fe2df916f095e4068c49ba82d770a.

ceyhun committed:
Replace npm ci with npm install and add error message on git diff

ceyhun committed:
Test if a package-lock.json diff is catched on npm install

ceyhun committed:
Revert "Test if a package-lock.json diff is catched on npm install"

This reverts commit d67a0c0059b9a4195b3a9d31a2288c127d2a73db.

ceyhun committed:
Remove error message on git diff as it's already checked by 'check-local-changes'

ceyhun committed:
Convert a remaining npm ci to npm install

ceyhun committed:
Remove metro assets from iOS build cache

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Revert "Disable node_modules caches"

This reverts commit 54d45619cbadc2ce07eeeec4e1901fc90ebb2ae7.

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Merge branch 'update/react-17' into rnmobile/try/upgrade-0-64

ceyhun committed:
Fix error in templates.test after React 17 upgrade

ceyhun committed:
Fix error in templates.test after React 17 upgrade

ceyhun committed:
Update react-native-linear-gradient and react-native-hsv-color-picker refs

ceyhun committed:
Update react-native-linear-gradient and react-native-hsv-color-picker refs

cameronvoell committed:
Update react-native-linear-gradient version

ceyhun committed:
Update react-native-hsv-color-picker ref

hypest committed:
Bump react-native-safe-area-context to v3.2.0

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

hypest committed:
Bump to include build fixes for react-native-safe-area-context

hypest committed:
Bump react-native-safe-area-context to include using RN 0.64

hypest committed:
Bump react-native-slider to include using RN 0.64

hypest committed:
Bump react-native-get-random-values to include using RN 0.64

hypest committed:
Bump react-native-masked-view to include using RN 0.64

hypest committed:
Bump react-native-screens to include using RN 0.64

hypest committed:
Bump react-native-reanimated to include using RN 0.64

hypest committed:
Bump react-native-prompt-android to include using RN 0.64

youknowriad committed:
Fix remaining e2e test failure

hypest commented:
I restarted the "Compressed Size / Check (14) (pull_request)" job even though it was successful already, to see if it still succeeds. That job fails in https://github.com/WordPress/gutenberg/pull/32517 which is not clear to me yet why.

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun committed:
Merge branch 'update/react-17' into rnmobile/try/upgrade-0-64

ceyhun committed:
Changes to package-lock.json after running npm install

ceyhun committed:
Changes to Podfile.lock after running npm run native ios

ceyhun committed:
Trigger CI

gziolo commented on .github/workflows/create-block.yml:
```diff
@@ -39,5 +39,5 @@ jobs:
             - name: npm install, build, format and lint
               run: |
-                  npm ci
+                  npm install
```
Can you explain why is `npm install` necessary with React 0.64?

ceyhun replied:
Sorry, I should've mentioned this in the description, will add it there shortly after.

`npm ci` was throwing `cb() never called!` errors randomly both on Linux and macOS on GitHub Actions. We're still not sure what's the reason. 
I tried various different combinations of node, npm, caching, etc. also debugging manually. Those attempts can be traced in some of [these commits](https://github.com/WordPress/gutenberg/commits/rnmobile/try/upgrade-0-64?after=436975285c0002e08cc33b62662e7549eff1e574+34&author=ceyhun&branch=rnmobile%2Ftry%2Fupgrade-0-64). Only thing that made the `cb() never called!` errors go away so far was using `npm install` instead of `npm ci`. I believe this might be an npm bug and hope it will be fixed with npm@7.

gziolo replied:
@youknowriad and @desrosj, do you know how to make those GitHub actions work by keeping `npm ci` that should be faster?

youknowriad replied:
I'm not sure this change is what caused the test to start passing, @desrosj made a change on trunk to improve the situation for these failures already. I think we still have some cases left but less than before.

hypest commented on packages/components/src/mobile/picker/index.android.js:
```diff
@@ -55,7 +55,7 @@ export default class Picker extends Component {
 
 		return options.map( ( option ) => (
-			<View key={ `${ option.label }-${ option.value }` }>
+			<Fragment key={ `${ option.label }-${ option.value }` }>
```
👋 @ceyhun , looks like [there's this PR](https://github.com/WordPress/gutenberg/pull/27326) that intentionally replaced the Fragment for a View. Do you think we should keep the View?

ceyhun replied:
In that PR the short syntax of Fragment was used like `<> </>`, which didn't support keys and maybe they were not aware of the [keyed Fragments](https://reactjs.org/docs/fragments.html#keyed-fragments). And also another issue that was solved in that PR, which was using `index` as a key and that was also in the commit in which this change was first introduced [here](https://github.com/WordPress/gutenberg/commit/d40168833531cbcc371c186864ebe3f91f44f14d#diff-9daa2ef947ebc2f02763111c4ffcac3a4f67c13f5c7ead89294ab24c10ed4a99). 
I think adding an extra `View` to the component tree shouldn't be necessary and `Fragment` should be fine.

gziolo commented on packages/components/src/scroll-lock/test/index.js:
```diff
@@ -41,6 +41,14 @@ describe( 'scroll-lock', () => {
+		// for this, yet. So for now, we'll use setTimeout to wait for cleanup.
+		//
+		// @see https://reactjs.org/blog/2020/08/10/react-v17-rc.html#effect-cleanup-timing
+		setTimeout( () => {
```
Unit tests have fake timers enabled so it might be a coincidence that `setTimeout` works as expected here. More details on the topic:

https://jestjs.io/docs/timer-mocks#advance-timers-by-time

I haven't tested it but it's probably more future-proof to code it as:

```js
jest.advanceTimersByTime( 1 ); // not sure about the time value
expectLocked( false );
```

hypest replied:
cc @vdwijngaert since I see that change introduced in [this commit](https://github.com/WordPress/gutenberg/commit/79b51c9d52e209fd5e51c8480be8962b5596378f) in the [React v17 upgrade PR](https://github.com/WordPress/gutenberg/pull/26847).

ceyhun replied:
I don't remember this test failing even before I merged in the React 17 upgrade branch, maybe it's because I had installed a more recent version of [@wojtekmaj/enzyme-adapter-react-17](https://github.com/wojtekmaj/enzyme-adapter-react-17) (0.6.2 instead of 0.3.1). I'll try reverting this commit altogether to see if it works fine.

ceyhun replied:
Reverting the commit didn't work :) But using `jest.advanceTimersByTime` instead worked fine. [`3e095a7` (#29118)](https://github.com/WordPress/gutenberg/pull/29118/commits/3e095a70589f5a3dc64a309d6159875bc742ef29)

gziolo replied:
I'm happy but surprised it worked on first try 🤣

gziolo commented on packages/react-native-bridge/.eslintrc.js:
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	extends: '../../.eslintrc.js',
+	settings: {
```
It should be moved to the overrides section for RN files:

https://github.com/WordPress/gutenberg/blob/eaa1be08e278a76774c040414c1691058caaa653/.eslintrc.js#L167-L179

This way we can control it from one place and avoid local ESLint config files.

gziolo commented on packages/react-native-aztec/.eslintrc.js:
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	extends: '../../.eslintrc.js',
+	settings: {
```
Same as in https://github.com/WordPress/gutenberg/pull/29118/files#r651667754.

hypest commented on packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:
```diff
@@ -559,9 +559,9 @@ public void setShouldDeleteEnter(final ReactAztecText view, boolean shouldDelete
 
     @Override
-    public void receiveCommand(final ReactAztecText parent, int commandType, @Nullable ReadableArray args) {
+    public void receiveCommand(final ReactAztecText parent, String commandType, @Nullable ReadableArray args) {
```
Looks like not using the `receiveCommand(..., int ..., ...)` variant anymore ([which is deprecated alright](receiveCommand)) would let us also remove [the getCommandsMap override](https://github.com/WordPress/gutenberg/blob/436975285c0002e08cc33b62662e7549eff1e574/packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java#L554). No need to do it in this PR, just mentioning it for our future selves. We should open a ticket about it after merging the RN 0.64 upgrade.

ceyhun replied:
Sounds good 👍 I have limited knowledge around here so I only applied the minimum changes necessary to make this work, but I agree that this file might need some further refactoring.

ceyhun replied:
Created an issue for that: https://github.com/wordpress-mobile/gutenberg-mobile/issues/3635

hypest reviewed (approved):
Had a look at the native mobile changes and look good to me! I've only left some non-blocking comments.

Also, I didn't particularly review the iOS side changes but I trust our (successful) testing so far for that. At this point, and since the upgrade effort is already monumental, takes long time and is hard to keep up-to-date, I'm fine with optimistically merging and we can fix any issues afterwards.

That said, since a few folks are doing a pass on the web-side of things (I see @gziolo's for example), let's wait for those review passes to conclude as well before merging.

gziolo commented:
From my perspective, the only concern for merging this PR is `npm ci` replaced in `npm install` in GitHub actions. It's something that we could change to move forward but we need to have a clear idea why it fails and how we can resolve it.

ceyhun committed:
Replace setTimeout with jest.advanceTimersByTime

ceyhun committed:
Move local eslint config to top-level overrides section

youknowriad commented:
> From my perspective, the only concern for merging this PR is npm ci replaced in npm install in GitHub actions.

Can we just revert that and see, I think the main problem for these failures is independent from the upgrade.

ceyhun committed:
Replace npm install with npm ci on CI

youknowriad commented:
Note that the two failures we're seeing now (performance and post title) are unrelated for this PR and exists on trunk and we're working on them in #32686

gziolo reviewed (approved):
Great work, assuming that `npm ci` works the same way as other PRs at the moment, this patch looks good to go.

Amazing teamwork everyone involved to get RN 0.64 and React 17 upgrades working 🎉 👏🏻

ceyhun committed:
Revert workflows that were originally using npm install

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64

ceyhun commented:
> Note that the two failures we're seeing now (performance and post title) are unrelated for this PR and exists on trunk and we're working on them in #32686

Great to hear that 😅 I've pushed 2 commits (https://github.com/WordPress/gutenberg/pull/29118/commits/85c4331a944a6db374500f9faaa418ca8b741138, https://github.com/WordPress/gutenberg/pull/29118/commits/e8cd4af6dfc63f64e2fe20b862c6a99b35ab1b1a) replacing `npm install` with `npm ci` to how they were before. Didn't see any `cb() never called!` errors, but I fear that might be related to `~/.npm` cache still being in place and maybe the error will return when a new change to `package-lock.json` that makes the cache invalid. In any case I think we can go back to `npm install` or try to find another solution if that ever happens.

I'll get one final merge from `trunk` to fix remaining checks as https://github.com/WordPress/gutenberg/pull/32686 seems to have been merged to `trunk`.

ceyhun committed:
Merge branch 'trunk' into rnmobile/try/upgrade-0-64
